### PR TITLE
Replace std::vector with std::array when possible

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
This is part of #15291.

I reviewed:
`Framework/Algorithms/`
`Framework/API/`
`Framework/Crystal/`

I found just a few places where we can easily replace `std::vector` by `std::array`. In many of the cases where we have an array of known size (see for instance [here](https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/SANSDirectBeamScaling.cpp#L171-L174) or [here](https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/UnwrapMonitor.cpp#L415-L419))  we have to use `std::vector` as it is then used as an input/output property, or input/output to a method that requires a vector.

Tester: just review code changes. No need to include this in the release notes.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
Changes appear to be good and not affecting functionality.
:shipit:

**Comment by FedeMPouzols** [comment]
:+1: 
This is a tricky ticket. I'd guess there are potentially many places where `std::array` could replace `std::vector` but that will take some time.
Good to use `std::array` when possible so we have examples in the code and `std::vector`s are not used everywhere as only alternative. I'd guess the performance difference will be noticeable only in tight loops, if any.

This is somewhat related to #15341 in that it also introduces initializer lists.
There's the (minor style) question whether to prefer `std::vector<int> v =  { 1, 2, 3 };` or `std::vector<int> v{ 1, 2, 3 };`

File: Framework/Algorithms/test/FindPeakBackgroundTest.h [hunk 1]
```diff
@@ -67,31 +67,14 @@ class FindPeakBackgroundTest : public CxxTest::TestSuite {
...
-    data.push_back(42);
-    data.push_back(77);
-    data.push_back(67);
-    data.push_back(33);
-    data.push_back(27);
-    data.push_back(20);
-    data.push_back(9);
-    data.push_back(2);
+
+    const size_t size = 20;
```
Comment (FedeMPouzols) [inline Framework/Algorithms/test/FindPeakBackgroundTest.h]: This is a test with fixed inputs so `size = 20` shouldn't be a problem, but I'd tend to prefer getting it from the array after it is declared: `const size_t size = data.size()`. Especially in the next changes, where the number of items is large and it's easy to make a mistake.

Comment (raquelalvarezbanos) [inline Framework/Algorithms/test/FindPeakBackgroundTest.h]: Thanks Federico, you are right that in big arrays it is easy to make a mistake, I'll use `data.size()` instead in future changes.

**Pull request merged**